### PR TITLE
feat(adapter): Azure Table Storage DB Adapter

### PR DESCRIPTION
**Pull request by nikitaclicks**
> _NOTE_:
>
> - It's a good idea to open an issue first to discuss potential changes.
> - Please make sure that you are _NOT_ opening a PR to fix a potential security vulnerability. Instead, please follow the [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md) to disclose the issue to us confidentially.

## ☕️ Reasoning

Add a DB adapter for Azure Table Storage.

## 🧢 Checklist

- [x] Documentation
- [x] Tests
- [x] Ready to be merged

## 📌 Resources

- [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md)
- [Contributing guidelines](https://github.com/nextauthjs/.github/blob/main/CONTRIBUTING.md)
- [Code of conduct](https://github.com/nextauthjs/.github/blob/main/CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

### CODE REVIEW

File: packages/adapter-table-storage/README.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+<p align="center">
+   <br/>
+   <a href="https://authjs.dev" target="_blank"><img height="64px" src="https://authjs.dev/img/logo/logo-sm.png" /></a>&nbsp;&nbsp;&nbsp;&nbsp;<img height="64px" src="./logo.svg" />
+   <h3 align="center"><b>Azure Table Storage Adapter</b> - NextAuth.js</h3>
```
Comment (ndom91) [inline packages/adapter-table-storage/README.md:4]:
```suggestion
   <h3 align="center"><b>Azure Table Storage Adapter</b> - Auth.js</h3>
```

**Comment by ndom91** [comment]
So generally this looks pretty good. Tests are passing for me locally. There are a good deal of formatting issues though, could you run eslint/prettier over all your `.ts` files? (for example: https://github.com/nextauthjs/next-auth/pull/6494/files#diff-500dc5f7fc44c7ab6c35911261585f084d53d0df101fcf6628f2724885fa0a6cR1-R3)

**Comment by nikitaclicks** [comment]
@ndom91 @linyiru @youpy @jrasanen hello team! Is there any chance for this to get reviewed this week?

**Comment by ThangHuuVu** [comment]
hi @nikitaclicks, I copy your changes to #8708, here are some changes that I made:
- rename it to `@auth/azure-tables`
- Tweak the logic of keys to use user by ID by default and use userByEmail for email, to make it consistent with other KV adapters
- Doc & misc
- Tests are passing so I merged it! Thanks for the contribution. I'm closing this one.

**Pull request closed without merging**